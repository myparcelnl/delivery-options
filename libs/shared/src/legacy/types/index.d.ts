import {type Vue} from 'vue/types/vue';

/**
 * @deprecated
 * @deprecated
 */
declare namespace MyParcel {
  /** @deprecated */
  type Environment = 'dev' | 'staging' | 'acceptance' | 'prod';

  /** @deprecated */
  type CarrierName = 'postnl' | 'bpost' | 'dpd' | 'dhl' | 'ups' | 'dhlforyou' | 'dhlparcelconnect' | 'dhleuroplus';

  /** @deprecated */
  type CarrierID = 1 | 2 | 4 | 6 | 8 | 9 | 10 | 11;

  /** @deprecated */
  type CarrierNameOrId = CarrierName | CarrierID;

  /** @deprecated */
  type Platform = 'myparcel' | 'belgie' | 'flespakket';

  /** @deprecated */
  type CarrierIdentifier = CarrierName | `${CarrierName}:${number}`;

  /**
   * @deprecated
   * @see https://myparcelnl.github.io/api/#6_A_1
   */
  type PackageType = 'package' | 'mailbox' | 'digital_stamp' | string;

  /**
   * @deprecated
   * @see https://myparcelnl.github.io/api/#6_A_2
   */
  type DeliveryType = 'morning' | 'standard' | 'evening' | 'pickup';

  /**
   * @deprecated
   * @see https://myparcelnl.github.io/api/#6_A_3
   */
  type ShipmentOptionName =
    | 'cooled_delivery'
    | 'large_format'
    | 'only_recipient'
    | 'signature'
    | 'return'
    | 'same_day_delivery';
}

/** @deprecated */

declare namespace MyParcelDeliveryOptions {
  /** @deprecated */
  type DeliveryType = MyParcel.DeliveryType | 'sameDay';

  /**
   * @deprecated
   * Configuration object supplied by the platform.
   */
  /** @deprecated */
  interface Configuration {
    address: Address;
    config: Config;
    strings: Strings;
  }

  /**
   * @deprecated
   * Address object from the external platform.
   */
  /** @deprecated */
  interface Address {
    cc: string;
    city?: string;
    number: string | number;
    postalCode: string;
  }

  /**
   * @deprecated
   * Strings object from the external platform.
   */
  /** @deprecated */
  interface Strings {
    addressNotFound?: string;
    // BE only
    beDeliveryStandardTitle?: string;
    beDeliveryTitle?: string;
    cc?: string;
    city?: string;
    closed?: string;
    deliveryEveningTitle?: string;
    deliveryMorningTitle?: string;
    deliveryStandardTitle?: string;
    deliveryTitle?: string;
    /**
     * @deprecated
     * @see src/config/errorConfig.js
     */
    error3212?: string;
    error3224?: string;
    error3505?: string;
    error3506?: string;
    error3728?: string;
    free?: string;
    from?: string;
    headerDeliveryOptions?: string;
    loadMore?: string;
    // NL only
    mondayDeliveryTitle?: string;
    number?: string;
    onlyRecipientTitle?: string;
    openingHours?: string;
    options?: string;
    packageTypeDigitalStamp?: string;
    packageTypeMailbox?: string;
    pickUp?: string;
    pickUpFrom?: string;
    pickupLocationsListButton?: string;
    pickupLocationsMapButton?: string;
    pickupTitle?: string;
    postalCode?: string;
    retry?: string;
    saturdayDeliveryTitle?: string;
    signatureTitle?: string;
    street?: string;
    wrongPostalCodeCity?: string;
    wrongnumberPostalCode?: string;
  }

  /**
   * @deprecated
   * Response from /delivery_options
   */
  /** @deprecated */
  interface DeliveryOption {
    date: Timestamp;
    possibilities: DeliveryPossibility[];
  }

  /**
   * @deprecated
   * Response from /pickup_locations
   */
  /** @deprecated */
  interface PickupLocation {
    address: {
      cc: string;
      city: string;
      number: string;
      postal_code: string;
      street: string;
    };
    location: {
      distance: string;
      latitude: string;
      location_code: string;
      location_name: string;
      longitude: string;
      phone_number: string;
      retail_network_id: string;
      opening_hours: {
        monday: StartEndDate[];
        tuesday: StartEndDate[];
        wednesday: StartEndDate[];
        thursday: StartEndDate[];
        friday: StartEndDate[];
        saturday: StartEndDate[];
        sunday: StartEndDate[];
      };
    };
    possibilities: PickupPossibility[];
  }

  /** @deprecated */
  interface FilterableOption {
    allow: boolean;
    items: string[];
  }

  /**
   * @deprecated
   * Configuration object from the external platform.
   */
  /** @deprecated */
  interface Config {
    allowDeliveryOptions?: boolean | FilterableOption;
    allowPickupLocations?: boolean | FilterableOption;
    // Feature toggles
    allowRetry?: boolean;
    allowShowDeliveryDate?: boolean;
    apiBaseUrl?: string;
    carrierSettings?: CarrierSettings;
    currency?: string;
    cutoffTime?: string;
    deliveryDaysWindow?: string | number;
    dropOffDays?: string;
    dropOffDelay?: string | number;
    locale?: string;
    // NL only
    mondayCutoffTime?: string;
    packageType?: MyParcel.PackageType;
    pickupLocationsDefaultView?: 'map' | 'list';
    // Can be JSON string or object.
    pickupLocationsMapTileLayerData?: string | MapTileLayerData;
    pickupShowDistance?: boolean;
    platform?: MyParcel.Platform;
    // BE only
    saturdayCutoffTime?: string;
  }

  /** @deprecated */
  type CarrierSettings = {
    [key in MyParcel.CarrierIdentifier]?: {
      allowEveningDelivery?: boolean;
      allowMorningDelivery?: boolean;
      allowOnlyRecipient?: boolean;
      allowSignature?: boolean;
      allowShowDeliveryDate?: boolean;

      // NL only
      allowSaturdayDelivery?: boolean;

      // BE only
      allowMondayDelivery?: boolean;

      priceEveningDelivery?: number;
      priceMorningDelivery?: number;
      priceOnlyRecipient?: number;
      pricePickup?: number;
      priceSameDayDelivery?: number;
      priceSignature?: number;
      priceStandardDelivery?: number;
    };
  };

  /** @deprecated */
  interface CarrierData {
    deliverCountries?: string[];
    deliveryEnabled?: boolean;
    id: MyParcel.CarrierID;
    identifier: MyParcel.CarrierIdentifier;
    image: string;
    label: string;
    name: MyParcel.CarrierName;
    pickupCountries?: string[];
    pickupEnabled?: boolean;
  }

  /** @deprecated */
  interface Timestamp {
    date: string;
    timezone: string;
    timezone_type: number;
  }

  /**
   * @deprecated
   * A start and end date object.
   */
  /** @deprecated */
  interface StartEndDate {
    end: Timestamp;
    start: Timestamp;
  }

  /** @deprecated */
  interface EnumSchema<Type> {
    enum: Type[];
    type: Type extends boolean ? 'boolean' : string;
  }

  /** @deprecated */
  interface ShipmentOption {
    name: MyParcel.ShipmentOptionName;
    schema: EnumSchema<boolean>;
  }

  /** @deprecated */
  interface DeliveryTimeFrame<Type = 'start' | 'end'> {
    date_time: Timestamp;
    type: Type;
  }

  /** @deprecated */
  interface DeliveryPossibility {
    collect_date?: any;
    delivery_time_frames: [DeliveryTimeFrame<'start'>, DeliveryTimeFrame<'end'>];
    package_type: string;
    shipment_options: ShipmentOption[];
    type: MyParcel.DeliveryType;
  }

  /** @deprecated */
  interface PickupPossibility {
    delivery_type_id: number;
    delivery_type_name: MyParcel.DeliveryType;
    moment: {
      start: Timestamp;
    };
  }

  /** @deprecated */
  interface FormConfig {
    enabled?: string;
    label?: string;
    name: string;
    options?: FormConfig[];
    price?: string;
    selected?: boolean;
  }

  /** @deprecated */
  interface FormEntry {
    choices?: FormEntryChoice[];
    component?: Vue;
    dependency?: FormEntryDependency;
    hidden?: boolean;
    loop?: boolean;
    name: string;
    pagination?: number;
    type?: 'radio' | 'select' | 'checkbox' | 'text' | string;
  }

  /** @deprecated */
  interface FormEntryChoice {
    disabled?: boolean;
    label?: string;
    name: string;
    plainLabel?: string;
    price?: number;
    selected?: boolean;
  }

  /** @deprecated */
  interface FormEntryDependency {
    name: string;
    parent: string;
    transform?: Function;
  }

  /** @deprecated */
  interface MapTileLayerData {
    attribution: string;
    maxZoom?: number;
    token?: string;
    url: string;
  }

  /** @deprecated */
  type CarrierDeliveryDependencies = Record<MyParcel.CarrierName, DeliveryDependencies>;

  /** @deprecated */
  interface DeliveryDependencies {
    // ISO date string
    deliveryDate: Record<
      string,
      {
        deliveryMoment: DeliveryDependencyMoments;
      }
    >;
  }

  /** @deprecated */
  type DeliveryDependencyMoments = Record<DeliveryType, DeliveryDependencyMoment>;

  /** @deprecated */
  interface DeliveryDependencyMoment {
    moments: {
      start: string;
      end: string;
    };
    shipmentOptions: Record<MyParcel.ShipmentOptionName, EnumSchema<boolean>>;
  }

  /** @deprecated */
  interface DeliveryOptionsRequestParameters {
    carrier: MyParcel.CarrierIdentifier;
    cc: string;
    cutoff_time?: string;
    deliverydays_window?: number;
    dropoff_days?: string;
    dropoff_delay?: number;
    include?: 'shipment_options';
    monday_delivery?: boolean;
    number: number;
    package_type?: MyParcel.PackageType;
    platform: MyParcel.Platform;
    postal_code: string;
    saturday_delivery?: boolean;
  }
}

/** @deprecated */
declare module 'MyParcel' {
  export = MyParcel;
}

/** @deprecated */
declare module 'MyParcelDeliveryOptions' {
  export = MyParcelDeliveryOptions;
}
