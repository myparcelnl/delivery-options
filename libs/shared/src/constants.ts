import {PlatformName} from '@myparcel/constants';

export enum SubscriptionType {
  Required = 1,
  Never = 0,
  Optional = -1,
}

export const DEFAULT_PLATFORM = PlatformName.MyParcel;

export const FORM_NAME_DELIVERY_OPTIONS = 'deliveryOptions';

export enum CarrierSetting {
  AllowDeliveryOptions = 'allowDeliveryOptions',
  AllowEveningDelivery = 'allowEveningDelivery',
  AllowMondayDelivery = 'allowMondayDelivery',
  AllowMorningDelivery = 'allowMorningDelivery',
  AllowOnlyRecipient = 'allowOnlyRecipient',
  AllowPackageTypeDigitalStamp = 'allowPackageTypeDigitalStamp',
  AllowPackageTypeMailbox = 'allowPackageTypeMailbox',
  AllowPickupLocations = 'allowPickupLocations',
  AllowSameDayDelivery = 'allowSameDayDelivery',
  AllowSaturdayDelivery = 'allowSaturdayDelivery',
  /** @deprecated use ShowDeliveryDate instead */
  AllowShowDeliveryDate = 'allowShowDeliveryDate',
  AllowSignature = 'allowSignature',
  CutoffTime = 'cutoffTime',
  CutoffTimeSameDay = 'cutoffTimeSameDay',
  DeliveryDaysWindow = 'deliveryDaysWindow',
  DropOffDays = 'dropOffDays',
  DropOffDelay = 'dropOffDelay',
  FridayCutoffTime = 'fridayCutoffTime',
  PackageType = 'packageType',
  PriceEveningDelivery = 'priceEveningDelivery',
  PriceMondayDelivery = 'priceMondayDelivery',
  PriceMorningDelivery = 'priceMorningDelivery',
  PriceOnlyRecipient = 'priceOnlyRecipient',
  PricePackageTypeDigitalStamp = 'pricePackageTypeDigitalStamp',
  PricePackageTypeMailbox = 'pricePackageTypeMailbox',
  PricePickup = 'pricePickup',
  PriceSameDayDelivery = 'priceSameDayDelivery',
  PriceSaturdayDelivery = 'priceSaturdayDelivery',
  PriceSignature = 'priceSignature',
  PriceStandardDelivery = 'priceStandardDelivery',
  SaturdayCutoffTime = 'saturdayCutoffTime',
  ShowDeliveryDate = 'showDeliveryDate',
}

export enum ConfigSetting {
  ApiBaseUrl = 'apiBaseUrl',
  CarrierSettings = 'carrierSettings',
  Currency = 'currency',
  Locale = 'locale',
  PickupLocationsDefaultView = 'pickupLocationsDefaultView',
  PickupLocationsMapTileLayerData = 'pickupLocationsMapTileLayerData',
  PickupShowDistance = 'pickupShowDistance',
  Platform = 'platform',
  ShowDeliveryDate = 'showDeliveryDate',
  ShowPriceSurcharge = 'showPriceSurcharge',
  ShowPrices = 'showPrices',
}

export enum PickupLocationsView {
  Map = 'map',
  List = 'list',
}

export enum ComponentName {
  Button = 'Button',
  CheckboxGroup = 'CheckboxGroup',
  Checkbox = 'Checkbox',
  Radio = 'Radio',
  RadioGroup = 'RadioGroup',
  Select = 'Select',
  Text = 'Text',
}

export enum AddressField {
  Cc = 'cc',
  City = 'city',
  Number = 'number',
  PostalCode = 'postalCode',
  Street = 'street',
}

export enum OptionGroup {
  PackageType = 'packageType',
  Delivery = 'delivery',
  ShipmentOption = 'shipmentOption',
  Feature = 'feature',
  DropOff = 'dropOff',
}

export enum OptionType {
  String = 'string',
  Boolean = 'boolean',
  Number = 'number',
  Currency = 'currency',
  Date = 'date',
  Time = 'time',
  Select = 'select',
}

export const QUERY_KEY_CARRIERS = 'carriers';

export const QUERY_KEY_DELIVERY_OPTIONS = 'deliveryOptions';

export const QUERY_KEY_PICKUP_LOCATIONS = 'pickupLocations';
