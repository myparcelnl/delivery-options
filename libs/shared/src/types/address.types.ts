import {type AddressField} from '../enums';

interface BaseAddress {
  [AddressField.Cc]: string;
  [AddressField.City]: string;
  [AddressField.PostalCode]: string;
}

export interface DeliveryOptionsOldAddress extends BaseAddress {
  [AddressField.Number]: string | number;
}

export interface DeliveryOptionsAddress extends BaseAddress {
  [AddressField.Street]: string;
}

export interface ResolvedDeliveryOptionsAddress extends BaseAddress {
  [AddressField.Street]: string;
}
