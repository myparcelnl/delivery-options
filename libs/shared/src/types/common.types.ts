export type Translation = string;

export type ToRecord<T> = {
  [K in keyof T]: T[K];
};

export type MakeRequired<T, K extends keyof T> = Omit<T, K> & Required<Pick<T, K>>;

export type UnionExcept<T, U> = T extends U ? never : T;
