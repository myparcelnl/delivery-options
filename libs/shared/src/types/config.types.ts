import {type Component} from 'vue';
import {type RecursiveRequired} from '@myparcel/ts-utils';
import {
  type CarrierName,
  type DeliveryTypeName,
  type PackageTypeName,
  type ShipmentOptionName,
} from '@myparcel/constants';
import {type ComponentName, type OptionType, type PickupLocationsView} from '../enums';
import {type SupportedPlatformName} from './platform.types';
import {type DeliveryOptionsAddress} from './address.types';

interface MapTileLayerData {
  attribution: string;
  maxZoom?: number;
  token?: string;
  url: string;
}

export interface FilterableOption {
  allow: boolean;
  items: string[];
}

export type DeliveryOptionsStrings = Record<string, string>;

export type CarrierIdentifier = `${CarrierName}:${number}` | CarrierName;

export type TimestampString = `${number}:${number}` | string;

export type Price = number | null;

// interface BeCarrierSettings extends BaseCarrierSettings {
//   allowSaturdayDelivery?: boolean;
//   fridayCutoffTime?: TimestampString;
// }
//
// interface NlCarrierSettings extends BaseCarrierSettings {
//   allowMondayDelivery?: boolean;
//   saturdayCutoffTime?: TimestampString;
// }

export interface CarrierSettings {
  allowDeliveryOptions?: boolean | FilterableOption;
  allowEveningDelivery?: boolean | FilterableOption;
  allowMondayDelivery?: boolean;
  allowMorningDelivery?: boolean | FilterableOption;
  allowOnlyRecipient?: boolean;
  allowPackageTypeDigitalStamp?: boolean;
  allowPackageTypeMailbox?: boolean;
  allowPackageTypePackage?: boolean;
  allowPickupLocations?: boolean | FilterableOption;
  allowSameDayDelivery?: boolean;
  allowSaturdayDelivery?: boolean;
  /** @deprecated use ShowDeliveryDate instead */
  allowShowDeliveryDate?: boolean;
  allowSignature?: boolean;
  allowStandardDelivery?: boolean | FilterableOption;
  cutoffTime?: TimestampString;
  cutoffTimeSameDay?: TimestampString;
  deliveryDaysWindow?: number;
  dropOffDays?: number[];
  dropOffDelay?: number;
  fridayCutoffTime?: TimestampString;
  packageType?: PackageTypeName;
  priceEveningDelivery?: Price;
  priceMondayDelivery?: Price;
  priceMorningDelivery?: Price;
  priceOnlyRecipient?: Price;
  pricePackageTypeDigitalStamp?: Price;
  pricePackageTypeMailbox?: Price;
  pricePickup?: Price;
  priceSameDayDelivery?: Price;
  priceSaturdayDelivery?: Price;
  priceSignature?: Price;
  priceStandardDelivery?: Price;
  saturdayCutoffTime?: TimestampString;
  showDeliveryDate?: boolean;
}

export type CarrierSettingsObject = Partial<Record<CarrierIdentifier, CarrierSettings>>;

export interface DeliveryOptionsConfig extends CarrierSettings {
  apiBaseUrl?: string;
  carrierSettings?: CarrierSettingsObject;
  /**
   * Currency. Defaults to format of the browser.
   */
  currency?: string | undefined;
  initial?: Partial<DeliveryOptionsOutput>;

  /**
   * Locale. Defaults to the language of the browser.
   */
  locale?: string | undefined;

  pickupLocationsDefaultView?: PickupLocationsView;
  pickupLocationsMapTileLayerData?: string | MapTileLayerData;
  pickupShowDistance?: boolean;
  platform?: SupportedPlatformName;
  showDeliveryDate?: boolean;
  showPriceSurcharge?: boolean;
  showPrices?: boolean;
}

export interface DeliveryOptionsConfiguration {
  address: DeliveryOptionsAddress;
  config: DeliveryOptionsConfig;
  initial?: Partial<DeliveryOptionsOutput>;
  strings: DeliveryOptionsStrings;
}

export interface InputDeliveryOptionsConfiguration {
  address: DeliveryOptionsAddress;
  components: Partial<Record<ComponentName, Component>>;
  config: DeliveryOptionsConfig;
  initial?: Partial<DeliveryOptionsOutput>;
  strings?: DeliveryOptionsStrings;
}

export interface ResolvedDeliveryOptionsConfiguration {
  address: RecursiveRequired<DeliveryOptionsAddress>;
  config: RecursiveRequired<DeliveryOptionsConfig>;
  strings: RecursiveRequired<DeliveryOptionsStrings>;
}

interface ShipmentOptionsOutput {
  onlyRecipient?: boolean;
  signature?: boolean;
}

interface BaseOutput {
  carrier: CarrierIdentifier;
  date: string;
  deliveryType: DeliveryTypeName;
  isPickup: boolean;
  packageType: PackageTypeName;
  shipmentOptions: ShipmentOptionsOutput;
}

interface DeliveryOutput extends BaseOutput {
  deliveryType: DeliveryTypeName.Standard | DeliveryTypeName.Evening | DeliveryTypeName.Morning;
  isPickup: false;
}

interface PickupOutput extends BaseOutput {
  deliveryType: DeliveryTypeName.Pickup;
  isPickup: true;
  pickupLocation: {
    location: {
      latitude: number;
      longitude: number;
    };
    name: string;
    openingHours: string;
    pickupLocationCode: string;
    street: string;
    streetAdditionalInfo: string;
    city: string;
    postalCode: string;
  };
}

export type DeliveryOptionsOutput = DeliveryOutput | PickupOutput;

export type InternalOutput = {
  deliveryDate: string;
  /**
   * JSON encoded SelectedDeliveryMoment
   * @see SelectedDeliveryMoment
   */
  deliveryMoment: string;
  shipmentOptions?: ShipmentOptionName[];
};

export enum RelatedConfigOptionType {
  Allow = 'allow',
  Price = 'price',
  CutoffTime = 'cutoffTime',
}

export type RelatedConfigOption = {
  type: RelatedConfigOptionType;
  key: string;
};

export interface ConfigOption<T extends OptionType = OptionType> {
  key: string;
  parents?: string[];
  perCarrier?: boolean;
  related?: RelatedConfigOption[];
  type?: T;
  validators?: CustomValidator[];
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface CustomValidator<T1 = any, T2 = T1> {
  error: string;

  /**
   * Parse the input value.
   */
  parse?(value: T1): T2;

  /**
   * Validate the input value.
   */
  validate(value: T1): boolean;
}
