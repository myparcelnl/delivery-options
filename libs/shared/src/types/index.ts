export * from './address.types';
export * from './carriers.types';
export * from './common.types';
export * from './config.types';
export * from './element.types';
export * from './form.types';
export * from './input.types';
export * from './options.types';
export * from './platform.types';
export * from './query.types';
export * from './validator.types';
