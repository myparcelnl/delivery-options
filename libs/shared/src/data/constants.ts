import {DeliveryTypeName, PackageTypeName, PlatformName, ShipmentOptionName} from '@myparcel/constants';
import {type DropOffEntryObject} from '../types';
import {CarrierSetting} from './enums';

export const SUPPORTED_PLATFORMS = [PlatformName.MyParcel, PlatformName.SendMyParcel] as const;

export const SUPPORTED_DELIVERY_TYPES = [
  DeliveryTypeName.Standard,
  DeliveryTypeName.Evening,
  DeliveryTypeName.Morning,
  DeliveryTypeName.Pickup,
] as const;

export const SUPPORTED_SHIPMENT_OPTIONS = [ShipmentOptionName.OnlyRecipient, ShipmentOptionName.Signature] as const;

export const SUPPORTED_PACKAGE_TYPES = [
  PackageTypeName.Package,
  PackageTypeName.Mailbox,
  PackageTypeName.DigitalStamp,
  PackageTypeName.PackageSmall,
] as const;

export const DEFAULT_PLATFORM = PlatformName.MyParcel;

export const FORM_NAME_DELIVERY_OPTIONS = 'deliveryOptions';

export const REQUEST_KEY_CARRIERS = 'carriers';

export const REQUEST_KEY_DELIVERY_OPTIONS = 'deliveryOptions';

export const REQUEST_KEY_PICKUP_LOCATIONS = 'pickupLocations';

export const DROP_OFF_WEEKDAY = 'weekday' satisfies keyof DropOffEntryObject;

export const DROP_OFF_CUTOFF_TIME = CarrierSetting.CutoffTime satisfies keyof DropOffEntryObject;

export const DROP_OFF_SAME_DAY_CUTOFF_TIME = CarrierSetting.CutoffTimeSameDay satisfies keyof DropOffEntryObject;

export const DAY_MONDAY = 1;

export const DAY_TUESDAY = 2;

export const DAY_WEDNESDAY = 3;

export const DAY_THURSDAY = 4;

export const DAY_FRIDAY = 5;

export const DAY_SATURDAY = 6;

export const DAY_SUNDAY = 0;

export const DAYS_IN_WEEK = 7;

export const NBSP = '\u00A0';

export const CARRIER_IDENTIFIER_SEPARATOR = ':';

export const API_DATE_FORMAT = 'y-MM-dd kk:mm:ss';

export const DEFAULT_TIMEZONE_TYPE = 3;

export const DEFAULT_TIMEZONE = 'Europe/Amsterdam';

export const DELIVERY_TIMEFRAME_TYPE_START = 'start';

export const DELIVERY_TIMEFRAME_TYPE_END = 'end';
