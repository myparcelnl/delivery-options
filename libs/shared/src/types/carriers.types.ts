import {type Carrier} from '@myparcel/sdk';
import {type CarrierName} from '@myparcel/constants';
import {
  type SupportedDeliveryTypeName,
  type SupportedPackageTypeName,
  type SupportedShipmentOptionName,
} from './platform.types';
import {type CarrierIdentifier, type ConfigKey} from './config.types';

export interface CarrierObject {
  identifier: CarrierIdentifier;
  name: CarrierName;
}

export interface CarrierWithIdentifier extends Carrier {
  identifier: CarrierIdentifier;
}

export type FullCarrier = CarrierWithIdentifier & {
  hasDeliveryInCountry(countryCode: string): boolean;
  hasDeliveryType(deliveryType: SupportedDeliveryTypeName): boolean;
  hasFeature(feature: ConfigKey): boolean;
  hasPackageType(packageType: SupportedPackageTypeName): boolean;
  hasPickupInCountry(countryCode: string): boolean;
  hasShipmentOption(shipmentOption: SupportedShipmentOptionName): boolean;
};
