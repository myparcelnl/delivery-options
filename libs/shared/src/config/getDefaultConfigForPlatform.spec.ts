import {describe, expect, it} from 'vitest';
import {PlatformName} from '@myparcel/constants';
import {type SupportedPlatformName} from '../types';
import {getDefaultConfigForPlatform} from './getDefaultConfigForPlatform';

describe('getDefaultConfigForPlatform', () => {
  it.each([PlatformName.MyParcel, PlatformName.SendMyParcel] satisfies SupportedPlatformName[])(
    'should return the correct default config for platform %s',
    (platform) => {
      const config = getDefaultConfigForPlatform(platform);

      expect(config).toMatchSnapshot();
    },
  );
});
