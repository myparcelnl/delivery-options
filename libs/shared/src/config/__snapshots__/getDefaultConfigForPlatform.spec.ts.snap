// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDefaultConfigForPlatform > should return the correct default config for platform belgie 1`] = `
{
  "allowDeliveryOptions": true,
  "allowEveningDelivery": true,
  "allowMondayDelivery": true,
  "allowMorningDelivery": true,
  "allowOnlyRecipient": true,
  "allowPickupLocations": true,
  "allowSameDayDelivery": true,
  "allowSignature": true,
  "apiBaseUrl": "https://api.myparcel.nl",
  "carrierSettings": {},
  "currency": "EUR",
  "cutoffTime": "17:00",
  "cutoffTimeSameDay": "09:30",
  "deliveryDaysWindow": 7,
  "dropOffDays": [
    1,
    2,
    3,
    4,
    5,
  ],
  "dropOffDelay": 0,
  "initial": {},
  "locale": "en-US",
  "packageType": "package",
  "pickupLocationsDefaultView": "list",
  "pickupLocationsMapTileLayerData": "{\\"url\\":\\"https://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png\\",\\"attribution\\":\\"&copy; <a href=\\\\\\"https://www.openstreetmap.org/copyright\\\\\\">OpenStreetMap</a> contributors, Tiles style by <a href=\\\\\\"https://www.hotosm.org/\\\\\\" target=\\\\\\"_blank\\\\\\">Humanitarian OpenStreetMap Team</a> hosted by <a href=\\\\\\"https://openstreetmap.fr/\\\\\\" target=\\\\\\"_blank\\\\\\">OpenStreetMap France</a>\\",\\"maxZoom\\":19}",
  "pickupShowDistance": false,
  "platform": "belgie",
  "showDeliveryDate": true,
  "showPriceSurcharge": false,
  "showPrices": true,
}
`;

exports[`getDefaultConfigForPlatform > should return the correct default config for platform myparcel 1`] = `
{
  "allowDeliveryOptions": true,
  "allowEveningDelivery": true,
  "allowMondayDelivery": true,
  "allowMorningDelivery": true,
  "allowOnlyRecipient": true,
  "allowPickupLocations": true,
  "allowSameDayDelivery": true,
  "allowSignature": true,
  "apiBaseUrl": "https://api.myparcel.nl",
  "carrierSettings": {},
  "currency": "EUR",
  "cutoffTime": "17:00",
  "cutoffTimeSameDay": "09:30",
  "deliveryDaysWindow": 7,
  "dropOffDays": [
    1,
    2,
    3,
    4,
    5,
  ],
  "dropOffDelay": 0,
  "initial": {},
  "locale": "en-US",
  "packageType": "package",
  "pickupLocationsDefaultView": "map",
  "pickupLocationsMapTileLayerData": "{\\"url\\":\\"https://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png\\",\\"attribution\\":\\"&copy; <a href=\\\\\\"https://www.openstreetmap.org/copyright\\\\\\">OpenStreetMap</a> contributors, Tiles style by <a href=\\\\\\"https://www.hotosm.org/\\\\\\" target=\\\\\\"_blank\\\\\\">Humanitarian OpenStreetMap Team</a> hosted by <a href=\\\\\\"https://openstreetmap.fr/\\\\\\" target=\\\\\\"_blank\\\\\\">OpenStreetMap France</a>\\",\\"maxZoom\\":19}",
  "pickupShowDistance": true,
  "platform": "myparcel",
  "showDeliveryDate": true,
  "showPriceSurcharge": false,
  "showPrices": true,
}
`;
